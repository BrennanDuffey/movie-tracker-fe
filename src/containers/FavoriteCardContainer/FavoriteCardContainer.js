import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SmallCard from '../SmallCard/SmallCard';
import butterflixMascot from '../../images/character.svg';

export const FavoriteCardContainer = ({ movies }) => {
  
  let smallCards = movies
    .filter(movie => movie.isFavorite)
    .map(movie=> (<SmallCard movie={movie} key={movie.id} />))
  return (
      <main className="CardContainer">
        <header>
          <div className="title">
            <img src={butterflixMascot} alt="butterflix-fairy-mascot"/>
            <h1>ButterFlix</h1>
          </div>
          <div className="search-icon">
            <i className="fas fa-search"></i>
          </div>
        </header>
        <div className="display-divider"></div>
        <section className="cards-display">
          {smallCards}
        </section>
      </main>
  )
}

FavoriteCardContainer.propTypes={
  dispatch: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,
  movies: PropTypes.array,
  user: PropTypes.object
}

const mapStateToProps = (state) => ({
  movies: state.movies,
  user: state.user
});

export default connect(mapStateToProps)(FavoriteCardContainer);