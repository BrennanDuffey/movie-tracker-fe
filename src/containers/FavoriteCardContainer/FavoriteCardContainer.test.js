import {FavoriteCardContainer} from './FavoriteCardContainer';
import React from 'react';
import {shallow} from 'enzyme';

let mockMovies = [
  { title: "movie", poster: "http", backdrop: "http", summary: "words",
    genres: ["action"], release: 2000, rating: 9.9, id: 69, isFavorite: true
  },
  { title: "nextMovie", poster: "https", backdrop: "https", summary: "MoreWords",
    genres: ["drama"], release: 2001, rating: 9.2, id: 69, isFavorite: true
  }
];

describe('FavoriteCardContainer', ()=>{
  let wrapper;

  it('should match snapshot', ()=>{
    wrapper = shallow(<FavoriteCardContainer movies={mockMovies} />);

    expect(wrapper).toMatchSnapshot();
  })
})
