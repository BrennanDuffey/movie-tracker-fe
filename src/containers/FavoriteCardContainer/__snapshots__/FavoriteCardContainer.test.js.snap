// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCardContainer should match snapshot 1`] = `
<main
  className="CardContainer"
>
  <header>
    <div
      className="title"
    >
      <img
        alt="butterflix-fairy-mascot"
        src="character.svg"
      />
      <h1>
        ButterFlix
      </h1>
    </div>
    <div
      className="search-icon"
    >
      <i
        className="fas fa-search"
      />
    </div>
  </header>
  <div
    className="display-divider"
  />
  <section
    className="cards-display"
  >
    <ConnectFunction
      key="69"
      movie={
        Object {
          "backdrop": "http",
          "genres": Array [
            "action",
          ],
          "id": 69,
          "isFavorite": true,
          "poster": "http",
          "rating": 9.9,
          "release": 2000,
          "summary": "words",
          "title": "movie",
        }
      }
    />
    <ConnectFunction
      key="69"
      movie={
        Object {
          "backdrop": "https",
          "genres": Array [
            "drama",
          ],
          "id": 69,
          "isFavorite": true,
          "poster": "https",
          "rating": 9.2,
          "release": 2001,
          "summary": "MoreWords",
          "title": "nextMovie",
        }
      }
    />
  </section>
</main>
`;
